dev-utils/webdriver.js

```diff
@@ -233,94 +233,71 @@ function getWebdriveBaseConfig(
   return baseConfig
 }
 
-function waitForApmServerCalls(errorCount = 0, transactionCount = 0) {
-  const { name, version } = getBrowserInfo()
-  console.log(
-    `Waiting for minimum ${errorCount} Errors and ${transactionCount} Transactions in`,
-    name,
-    version
-  )
-  const serverCalls = browser.executeAsync(
-    function(errorCount, transactionCount, done) {
-      var apmServerMock = window.elasticApm.serviceFactory.getService(
-        'ApmServer'
-      )
-
-      function checkCalls() {
-        var serverCalls = apmServerMock.calls
-
-        var validCalls = true
-
-        if (errorCount) {
-          validCalls =
-            validCalls &&
-            serverCalls.sendErrors &&
-            serverCalls.sendErrors.length >= errorCount
-        }
-        if (transactionCount) {
-          validCalls =
-            validCalls &&
-            serverCalls.sendTransactions &&
-            serverCalls.sendTransactions.length >= transactionCount
-        }
-
-        if (validCalls) {
-          console.log('calls', serverCalls)
-          var promises = []
+function isPageLoaded() {
+  return browser.execute(() => {
+    return document.readyState === 'complete'
+  })
+}
 
-          if (serverCalls.sendErrors) {
-            promises = promises.concat(
-              serverCalls.sendErrors.map(function(s) {
-                return s.returnValue
-              })
-            )
-          }
-          if (serverCalls.sendTransactions) {
-            promises = promises.concat(
-              serverCalls.sendTransactions.map(function(s) {
-                return s.returnValue
-              })
-            )
-          }
+function waitForApmServerCalls(eventsCount = 0) {
```

Comment:
I think in the tests we want to differentiate between different even types (error, or transactions), for example in some tests we only care about transactions and we want to ignore the any number of errors.

Reply:
Its differentiated in the Tests in this way 

```js
const serverCalls = waitForServerCalls(2)
const { transactions,  errors } = serverCalls.sendEvents
```